pkg/cmd/roachtest/tests/sysbench.go

```diff
@@ -185,12 +185,21 @@ func runSysbench(ctx context.Context, t test.Test, c cluster.Cluster, opts sysbe
 	var start time.Time
 	runWorkload := func(ctx context.Context) error {
 		t.Status("preparing workload")
-		c.Run(ctx, option.WithNodes(c.WorkloadNode()), opts.cmd(true /* haproxy */)+" prepare")
+		cmd := opts.cmd(useHAProxy /* haproxy */)
+		result, err := c.RunWithDetailsSingleNode(ctx, t.L(), option.WithNodes(c.WorkloadNode()), cmd+" prepare")
+		if err != nil {
+			return err
+		} else if strings.Contains(result.Stdout, "FATAL") {
```

Comment:
Think this should be okay to do, looks like sysbench [doesn't do any retry logic](https://github.com/akopytov/sysbench/blob/de18a036cc65196b1a4966d305f33db3d8fa6f8e/src/lua/oltp_common.lua#L90-L93) so if we get just one FATAL we know setup for sure failed.